contracts/ColonyFunding.sol

```diff
@@ -166,44 +167,37 @@ contract ColonyFunding is ColonyStorage, DSMath {
     return nonRewardPotsTotal[_token];
   }
 
-  function getGlobalRewardPayoutCount() public view returns (uint256) {
-    return globalRewardPayoutCount;
-  }
-
-  function getUserRewardPayoutCount(address _user) public view returns (uint256) {
-    return userRewardPayoutCount[_user];
-  }
-
   function startNextRewardPayout(address _token) public auth {
+    ITokenLocking tokenLocking = ITokenLocking(IColonyNetwork(colonyNetworkAddress).getTokenLocking());
+    uint256 totalLockCount = tokenLocking.lockToken(address(token));
+
     require(!activeRewardPayouts[_token], "colony-reward-payout-token-active");
 
     uint256 totalTokens = sub(token.totalSupply(), token.balanceOf(address(this)));
     require(totalTokens > 0, "colony-reward-payout-invalid-total-tokens");
 
     activeRewardPayouts[_token] = true;
-    globalRewardPayoutCount += 1;
-
-    //TODO: Lock everyones tokens
 
-    rewardPayoutCycles[globalRewardPayoutCount] = RewardPayoutCycle(
+    rewardPayoutCycles[totalLockCount] = RewardPayoutCycle(
       IColonyNetwork(colonyNetworkAddress).getReputationRootHash(),
       totalTokens,
       pots[0].balance[_token],
       _token,
       block.timestamp
     );
 
-    emit RewardPayoutCycleStarted(globalRewardPayoutCount);
+    emit RewardPayoutCycleStarted(totalLockCount);
   }
 
   function claimRewardPayout(uint256 _payoutId, uint256[7] _squareRoots, uint256 _userReputation, uint256 _totalReputation) public {
     RewardPayoutCycle memory payout = rewardPayoutCycles[_payoutId];
+    // Checking if payout is active
     require(block.timestamp - payout.blockTimestamp <= 60 days, "colony-reward-payout-not-active");
-    require(_payoutId - userRewardPayoutCount[msg.sender] == 1, "colony-reward-payout-bad-id");
```

Comment:
No, will add it.
Since we only allow claiming next in 'queue', we could also not pass `_payoutId` and just take current users token lock count and add `1`. But i guess its not significantly different that this